### ISSUE TITLE
MenuItems with IsCheckable and StaysOpenOnClick set to true do not announce to Narrator when IsChecked is changed

### ISSUE THREAD

**Issue by mslukebo** [dotnet/wpf#5237]
* .NET Core Version:  3.1
* Does the bug reproduce also in WPF for .NET Framework 4.8?: Unknown
* Is this bug related specifically to tooling in Visual Studio (e.g. XAML Designer, Code editing, etc...)? No
* Security issues and bugs should be reported privately, learn more via our [responsible disclosure guidelines]

 **Problem description:**

If you have a `MenuItem` with `IsCheckable=True` and `StaysOpenOnClick=True`, Windows Narrator does not announce when the `IsChecked` state is changed.
 
 **Actual behavior:**

Windows Narrator does not announce anything when `IsChecked` is changed (e.g. the item is clicked on or the user types "Enter" while focusing it).
 
 **Expected behavior:**

Windows Narrator should alert the user that the state has changed by saying "<MenuItem.Header> Checked" or "<MenuItem.Header> Unchecked"

**Comment by anjalisheel-wpf** [comment]
@[mslukebo](https://github.com/mslukebo) Could you please provide us sample app for the same

**Comment by mslukebo** [comment]
> @[mslukebo](https://github.com/mslukebo) Could you please provide us sample app for the same

I have provided repro steps in the original bug report, so it should be easy to recreate on a new WPF application.
